src/mscorlib/shared/System/ArgumentException.cs

```diff
@@ -64,12 +65,13 @@ public ArgumentException(String message, String paramName)
         protected ArgumentException(SerializationInfo info, StreamingContext context)
             : base(info, context)
         {
-            throw new PlatformNotSupportedException();
+            _paramName = info.GetString("ParamName");
```

Comment:
Are we certain it's OK for the field name to be different so long as we're serializing it explicitly like this? Maybe better to rename it to `m_paramName` with the usual "don't change me" comment, just to be sure.

Reply:
the field name doesn't matter at all.

Reply:
> the field name doesn't matter at all.

Right; once you implement ISerializable, fields aren't inspected.

Reply:
Ah, I see desktop already had cases where their field names differed. It must be totally safe. We can leave our names.
https://referencesource.microsoft.com/#mscorlib/system/badimageformatexception.cs,154